lib/backend/k8s/resources/ippool.go

```diff
@@ -41,7 +43,36 @@ func NewIPPoolClient(c *kubernetes.Clientset, r *rest.RESTClient) K8sResourceCli
 			Kind:       apiv3.KindIPPool,
 			APIVersion: apiv3.GroupVersionCurrent,
 		},
-		k8sListType:  reflect.TypeOf(apiv3.IPPoolList{}),
-		resourceKind: apiv3.KindIPPool,
+		k8sListType:      reflect.TypeOf(apiv3.IPPoolList{}),
+		resourceKind:     apiv3.KindIPPool,
+		versionconverter: IPPoolv1v3Converter{},
+	}
+}
+
+// IPPoolv1v3Converter implements VersionConverter interface.
+type IPPoolv1v3Converter struct{}
+
+// ConvertFromK8s converts v1 IPPool Resource to v3 IPPool resource
+func (c IPPoolv1v3Converter) ConvertFromK8s(inRes Resource) (Resource, error) {
+	ipp, ok := inRes.(*apiv3.IPPool)
+	if !ok {
+		return nil, fmt.Errorf("invalid type conversion")
+	}
+
+	// If IPIP field is not nil, then it means the resource has v1 IPIP data
+	// and we must convert it to v3equivalent data.
+	if ipp.Spec.IPIP != nil {
+		if !ipp.Spec.IPIP.Enabled {
+			ipp.Spec.IPIPMode = apiv3.IPIPModeNever
+		} else if ipp.Spec.IPIP.Enabled && ipp.Spec.IPIP.Mode == ipip.CrossSubnet {
```

Comment:
IPIP.Enabled is redundant.  You know it's true already.

Reply:
yeah added it for the clarity, but i can remove it